lib/web_ui/lib/src/engine/window.dart

```diff
@@ -123,12 +123,12 @@ class EngineFlutterWindow extends ui.SingletonFlutterWindow {
         return true;
       case 'routeUpdated': // deprecated
         await _useSingleEntryBrowserHistory();
-        browserHistory.setRouteName(arguments['routeName']);
+        browserHistory.setRouteName(arguments?['routeName']);
```

Comment:
setRouteName currently either asserts(routeName != null) or calls pushState. Need to update history.dart to check for null or do check at call site here and below on line 130

Reply:
It seems the code is already "handling" the case where `routeName` may be null:

https://github.com/flutter/engine/blob/d1c92dff94274f5e614302120e110aaae2a74eea/lib/web_ui/lib/src/engine/navigation/history.dart#L136

or:

https://github.com/flutter/engine/blob/d1c92dff94274f5e614302120e110aaae2a74eea/lib/web_ui/lib/src/engine/navigation/history.dart#L268

(In both cases, it seems purposefully built to fail at that point?)

Reply:
Correct. In both implementations it will assert. maybe make it explicity in second case as well with assert(routeName != null);

Reply:
Ah, my bad @ferhatb; I just realized that the second implementation handles null `routeName` values below:

https://github.com/flutter/engine/blob/d1c92dff94274f5e614302120e110aaae2a74eea/lib/web_ui/lib/src/engine/navigation/history.dart#L326-L335

(See how in line 334 it either picks the `routeName` passed in (as `path`) or defaults to `currentPath`.)

Reply:
Before this change, `routeUpdated` and `routeInformationUpdated` were never called with `null` arguments.

To fix this, I added a not-null assertion to the two old cases, to preserve the behavior of the old version of the method (with regards to those two messages).